OpenSim/Simulation/Test/testContactGeometry.cpp

```diff
@@ -339,16 +354,206 @@ void compareHertzAndMeshContactResults()
+    model.addContactGeometry(floor);
+
+    return model;
+}
+
+template <typename ContactType>
+void testIntermediateFrames() {
+
+    // Simulate the given model for one second and return the final state.
+    auto simulate = [](Model& model) -> SimTK::State {
+        // Initialize and set state.
+        SimTK::State& state = model.initSystem();
+        const auto& hinge = model.getJointSet().get("hinge");
+        const auto& coord = hinge.get_CoordinateSet()[0];
+        coord.setValue(state, 0.15 * SimTK::Pi);
+
+        // Integrate.
+        RungeKuttaMersonIntegrator integrator(model.getMultibodySystem());
+        integrator.setAccuracy(integ_accuracy);
+        Manager manager(model, integrator);
+        manager.setFinalTime(1.0);
+        manager.integrate(state);
+
+        return state;
+    };
+
+    // Obtain result from using a weld joint and another body to achieve the
+    // "intermediate frame."
+    SimTK::State stateWeld;
+    {
+        Model model = createBaseModel();
+        const auto& point = model.getBodySet().get("point");
+        auto* massless = new OpenSim::Body("massless", 0, Vec3(0), Inertia(0.0));
+        auto* weld = new OpenSim::WeldJoint("weld",
+                                   point, Vec3(0), Vec3(0),
+                                   // body is up 1m in the y direction.
+                                   *massless, Vec3(0, -1, 0), Vec3(0));
+        model.addBody(massless);
+        model.addJoint(weld);
+        addContactComponents<ContactType>(model, *massless, Vec3(0.5, 0, 0));
+        stateWeld = simulate(model);
+
+        // Make sure this model actually bounced; otherwise, the test is not
+        // meaningful.
+        SimTK_TEST(model.calcMassCenterVelocity(stateWeld)[1] > 0);
+    }
+
+    // Y offset accomplished with PhysicalOffsetFrame, X offset accomplished
+    // with ContactGeometry's location property.
+    SimTK::State stateIntermedFrameY;
+    {
+        Model model = createBaseModel();
+        auto* frame = new PhysicalOffsetFrame("intermed",
+                model.updBodySet().get("point"),
+                // Frame is up 1m in the y direction.
+                SimTK::Transform(Vec3(0, 1, 0)));
+        model.addFrame(frame);
+
+        addContactComponents<ContactType>(model, *frame, Vec3(0.5, 0, 0));
+        stateIntermedFrameY = simulate(model);
+
+        // Make sure this model actually bounced; otherwise, the test is not
+        // meaningful.
+        SimTK_TEST(model.calcMassCenterVelocity(stateIntermedFrameY)[1] > 0);
+    }
+
+    // The X and Y offsets are both accomplished with the PhysicalOffsetFrame.
+    SimTK::State stateIntermedFrameXY;
+    {
+        Model model = createBaseModel();
+        auto* frame = new PhysicalOffsetFrame("intermed",
+                model.updBodySet().get("point"),
+                // Frame is up 1m in the y direction.
+                SimTK::Transform(Vec3(0.5, 1, 0)));
```

Comment:
I was expecting you to use two `PhysicalOffsetFrames` to extend from the Y only offset from before. In which case you could just compare this answer to the weld and skip the Y only test. You could also combine rotational offsets. Since you are comparing to the welded mechanism anyway, you could use some random-like transform in both cases (Weld vs. Offset(s)).